### PR TITLE
Federation/key server publish

### PR DESCRIPTION
**Pull request by jendakol**
This is basically a client for upload to Key server

Part of #99

### CODE REVIEW

File: internal/functions/efgs/publish-to-key-server.go [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+		logger.Errorf("Error when getting token: %v", err)
+		return nil, err
+	}
+
+	hmacKey := make([]byte, 16)
+	_, _ = seededRand.Read(hmacKey)
+
+	certificate, err := getCertificate(ctx, *verificationServerConfig, keys, token, hmacKey)
+	if err != nil {
+		logger.Debugf("Error when getting certificate: %v", err)
```
Comment (dizider) [inline internal/functions/efgs/publish-to-key-server.go:83]: `logger.Errorf(...` shoud be better. This error is on the same level as error on line 77.

File: internal/functions/efgs/publish-to-key-server.go [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+		return nil, err
+	}
+
+	resp, err := publishKeys(ctx, *keyServerConfig, haid, keys, certificate, hmacKey)
+
+	if err != nil {
+		logger.Debugf("Error when publishing keys to Key server: %v", err)
```
Comment (dizider) [inline internal/functions/efgs/publish-to-key-server.go:90]: `logger.Errorf(...`

File: internal/functions/efgs/utils/verification.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	})
+
+	// Build the cleartext.
+	perKeyText := make([]string, 0, len(keys))
+	altPerKeyText := make([]string, 0, len(keys))
+	calculateAlt := true
+	for _, ek := range keys {
+		perKeyText = append(perKeyText,
+			fmt.Sprintf("%s.%d.%d.%d", ek.Key, ek.IntervalNumber, ek.IntervalCount, ek.TransmissionRisk))
+		altPerKeyText = append(altPerKeyText, fmt.Sprintf("%s.%d.%d", ek.Key, ek.IntervalNumber, ek.IntervalCount))
```
Comment (dizider) [inline internal/functions/efgs/utils/verification.go]: `altPerKeyText` is calculated but never used. `clearText` always contains joined `perKeyText`.

**Review by dizider — CHANGES_REQUESTED** [review]
When keys in `CalculateExposureKeysHMAC` does not contains `transmission risk` field, alternate key text must be used. Second variant is remove `altPerKeyText` and `calculateAlt`. Then when no value is assigned to `transsmision risk`, it must be presented as 0.

File: internal/functions/efgs/publish-to-key-server.go (hunk 1 above)
Comment (jendakol) [reply internal/functions/efgs/publish-to-key-server.go:83]: This is intentional - here it's logged as debug but it's propagated to the upper level, where it's logged again as failure as error. So, actually, lines 71 and 77 are IMHO wrong :-)  
Fixed now.

File: internal/functions/efgs/utils/verification.go (hunk 3 above)
Comment (jendakol) [reply internal/functions/efgs/utils/verification.go]: Oh. Refactoring mistake.

**Review by dizider — APPROVED** [review]
ok

**Pull request merged**